internal/cephfs/clone.go

```diff
@@ -188,37 +195,18 @@ func createCloneFromSnapshot(ctx context.Context, parentVolOpt, volOptions *volu
 	return nil
 }
 
-// CloneStatus represents the subvolume clone status.
-type CloneStatus struct {
-	Status struct {
-		State string `json:"state"`
-	} `json:"status"`
-}
-
-func getCloneInfo(ctx context.Context, volOptions *volumeOptions, cr *util.Credentials, volID volumeID) (CloneStatus, error) {
-	clone := CloneStatus{}
-	args := []string{
-		"fs",
-		"clone",
-		"status",
-		volOptions.FsName,
-		string(volID),
-		"--group_name",
-		volOptions.SubvolumeGroup,
-		"-m", volOptions.Monitors,
-		"-c", util.CephConfigPath,
-		"-n", cephEntityClientPrefix + cr.ID,
-		"--keyfile=" + cr.KeyFile,
-		"--format=json",
+func (vo *volumeOptions) getCloneInfo(ctx context.Context, volID volumeID) (cephFSCloneState, error) {
+	fsa, err := vo.conn.GetFSAdmin()
+	if err != nil {
+		util.ErrorLog(ctx, "could not get FSAdmin, can get clone status for volume %s with ID %s:", vo.FsName, string(volID), err)
+		return cephFSCloneError, err
 	}
-	err := execCommandJSON(
-		ctx,
-		&clone,
-		"ceph",
-		args[:]...)
+
+	cs, err := fsa.CloneStatus(vo.FsName, vo.SubvolumeGroup, string(volID))
 	if err != nil {
-		util.ErrorLog(ctx, "failed to get subvolume clone info %s(%s) in fs %s", string(volID), err, volOptions.FsName)
-		return clone, err
+		util.ErrorLog(ctx, "could not get clone state for volume %s with ID %s:", vo.FsName, string(volID), err)
```

Comment:
logging error is missing here?

Reply:
yeah, somehow that didn't get caught by some linter?